indexing-service/src/main/java/org/apache/druid/indexing/scheduledbatch/ScheduledBatchTaskManager.java

```diff
@@ -0,0 +1,298 @@
+    return new ScheduledBatchSupervisorSnapshot(
+        supervisorId,
+        manager.getSchedulerStatus(),
+        manager.getLastTaskSubmittedTime(),
+        manager.getNextTaskSubmissionTime(),
+        manager.getTimeUntilNextTaskSubmission(),
+        tasks.getSubmittedTasks(),
+        tasks.getCompletedTasks()
+    );
+  }
+
+  private void submitSqlTask(final String supervisorId, final ClientSqlQuery spec)
+      throws ExecutionException, InterruptedException
+  {
+    log.debug("Submitting a new task with spec[%s] for supervisor[%s].", spec, supervisorId);
+    final SqlTaskStatus taskStatus = FutureUtils.get(brokerClient.submitSqlTask(spec), true);
+    statusTracker.onTaskSubmitted(supervisorId, taskStatus);
+  }
+
+  private class SchedulerManager
+  {
+    private final String supervisorId;
+    private final String dataSource;
+    private final ClientSqlQuery spec;
+    private final CronSchedulerConfig cronSchedulerConfig;
+
+    private ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus status;
+
+    /**
+     * TODO: Note that the last task submitted per supervisor should eventually be persisted in the metadata store,
+     * along with any other scheduler state, so that the batch supervisor can recover gracefully and
+     * avoid missing task submissions during rolling restarts, etc.
+     */
+    private volatile DateTime lastTaskSubmittedTime;
+
+    private SchedulerManager(
+        final String supervisorId,
+        final String dataSource,
+        final CronSchedulerConfig cronSchedulerConfig,
+        final ClientSqlQuery spec
+    )
+    {
+      this.supervisorId = supervisorId;
+      this.dataSource = dataSource;
+      this.cronSchedulerConfig = cronSchedulerConfig;
+      this.spec = spec;
+    }
+
+    private synchronized void startScheduling()
+    {
+      if (jobsExecutor.isTerminated() || jobsExecutor.isShutdown()
+          || status == ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus.SCHEDULER_SHUTDOWN) {
+        return;
+      }
+
+      status = ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus.SCHEDULER_RUNNING;
+      final Duration timeUntilNextSubmission = getTimeUntilNextTaskSubmission();
+      if (timeUntilNextSubmission == null) {
+        log.info("No more tasks will be submitted for supervisor[%s].", supervisorId);
+        return;
+      }
+
+      jobsExecutor.schedule(
+          () -> {
+            try {
+              // Check status inside the runnable again before submitting any tasks
+              if (status == ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus.SCHEDULER_SHUTDOWN) {
+                return;
+              }
+              lastTaskSubmittedTime = DateTimes.nowUtc();
+              submitSqlTask(supervisorId, spec);
+              emitMetric("batchSupervisor/tasks/submit/success", 1);
+            }
+            catch (Exception e) {
+              emitMetric("batchSupervisor/tasks/submit/failed", 1);
```

Comment:
This metric need not be added in a new name bucket. Maybe use something like `task/scheduledBatch/submit/failed` instead.